odoo/models.py

```diff
@@ -4109,6 +4094,13 @@ def _load_records(self, data_list, update=False):
         records = self._load_records_create([data['values'] for data in to_create])
         for data, record in zip(to_create, records):
             data['record'] = record
+            # while at it, mark parents for which we'll need to create xids
+            values = data['values']
+            data['create_parent'] = {
+                parent_field
+                for parent_field in self._inherits.values()
+                if not values.get(parent_field)
+            }
```

Comment:
Wouldn't it be simpler to append more items to `imd_data_list` below?
I mean, to make all the items about parent records *explicit*?
This would totally remove the handling of parent records from `_update_xmlids`, and would isolate the logic here.

Reply:
> Wouldn't it be simpler to append more items to imd_data_list below?

Now that I look at it, it should be possible though it'll be more complicated than the above (though it'll also integrate some of `_update_xmlids`) as the "parent" ids need to be created before the "child" id.